source/menus/all_dungeons_saves_menu.cpp

```diff
@@ -0,0 +1,127 @@
+    {"nayru's pearl", NAYRUS_PEARL_AD, "at dri after drc for balloon swim to outset"},
+    {"enter tower of the gods", ENTER_TOTG_AD, "outside jabuun's cave for swim to sti/nti"},
+    {"tower of the gods", TOTG_AD, "start of totg"},
+    {"darknut", DARKNUT_AD, "darknut miniboss"},
+    {"after second statue", TOTG_AFTER_STATUE2_AD, "totg after returning second statue"},
+    {"gohdan", GOHDAN_BOSS_AD, "gohdan boss fight"},
+    {"puzzle skip", PUZZLE_SKIP_AD, "inside hyrule castle for puzzle skip"},
+    {"hyrule escape", HYRULE_ESCAPE_AD, "inside hyrule castle after master sword for hyrule escape"},
+    {"barrier skip", BARRIER_SKIP_AD, "outside hyrule castle for barrier skip"},
+    {"trial skip", TRIAL_SKIP_AD, "inside ganon's tower, in trials room for trials skip"},
+    {"boomerang skip", BOOMERANG_SKIP_AD, "in phantom ganon's room for boomerang skip"},
+    {"phantom ganon", PHANTOM_GANON_AD, "phantom ganon fight miniboss"},
+    {"enter helmaroc", ENTER_HELM_AD, "after phantom ganon fight for chest storage climb"},
+    {"hyrule 2 skip", HYRULE2_SKIP_AD, "in helm boss fight for ems cutscene skip"},
+    {"early wind god's aria", EARLY_WGA_AD, "at windfall after ff2 for swim to gale isle"},
+    {"power bracelets", FIRE_MOUNTAIN_AD, "at fire mountain for power braclets"},
+    {"early earth temple", EARLY_ET_AD, "at headstone for early earth temple"},
+    {"earth temple", ET_AD, "start of earth temple"},
+    {"song stone skip", SONGTONE_SKIP_AD, "in earth temple in song stone room"},
+    {"jalhalla", JALHALLA_AD, "jalhalla boss fight"},
+    {"iron boots", ICE_RING_AD, "at headstone for swim to ice ring for iron boots"},
+    {"makar", MAKAR_AD, "at ice ring for swim to forest haven for makar"},
+    {"enter wind temple", ENTER_WT_AD, "at forest haven for balloon swim to gale isle"},
+    {"wind temple", WT_AD, "start of wind temple"},
+    {"wizzrobe", WIZZROBE_AD, "wizzrobe miniboss"},
+    {"after hookshot", AFTER_HS_AD, "wind temple after hookshot"},
+    {"molgera", MOLGERA_AD, "molgera boss fight"},
+    {"swim to flight control platform", SWIM_2_FCP_AD, "at gale isle for swim to flight control platform"},
+    {"fcp layer manip", FCP_LAYER_MANIP_AD, "at flight control platform for layer manip dark portal"},
+    {"puppet ganon cutscene skip", PG_CS_AD, "in grand staircase for puppet ganon cutscene skip"},
+    {"puppet ganon skip", PG_SKIP_AD, "in puppet ganon room on layer 8 for puppet ganon skip"},
+    {"puppet ganon", PUPPET_GANON_AD, "puppet ganon fight if not doing pg skip"},
+    {"ganondorf", GANONDORF_AD, "ganondorf boss fight"},
+
+};
+
+void AllDungeonsSavesMenu::draw() {
+    cursor.setMode(Cursor::MODE_LIST);
+    cursor.move(0, LINE_NUM);
+
+    if (GZ_getButtonTrig(GZPad::B)) {
+        GZ_setMenu(GZ_PRACTICE_MENU);
+        return;
+    }
+
+    if (GZ_getButtonTrig(GZPad::A)) {
+        special AllDungeonsSpecials[] = {
+            special(MSS_AD, nullptr, nullptr),

```

Comment:
for every save special that uses nullptr for both callbacks, i strongly suggest removing them and testing to see how it performs. because a special with no callbacks is supposed to do nothing at all, meaning it's unnecessary data. if it's causing actual functional changes, then its a bug that needs to be fixed